Firestore/Source/API/FIRFirestore.mm

```diff
@@ -217,6 +217,13 @@ - (void)setSettings:(FIRFirestoreSettings *)settings {
                             "longer be changed. You can only set settings before calling any "
                             "other methods on a Firestore instance.");
     }
+    if (!settings.enableTimestampsInSnapshots) {
+      FSTWarn(
+          @"FirestoreSettings.enableTimestampsInSnapshots is set to false. This is legacy "
+           "behavior which will be removed in a future release. Please set "
+           "enableTimestampsInSnapshots to true and upgrade your code to support the new "
+           "behavior.");
```

Comment:
I'd phrase this more strongly and make it clearer what actions they need to take...  Something like:

```
The default behavior for NSDate objects stored in Firestore is going to change
AND YOUR APP MAY BREAK.

To hide this warning and ensure your app does not break, you need to add
the following code to your app before calling any other Cloud Firestore methods:

let db = Firestore.firestore()
let settings = db.settings
settings.enableTimestampsInSnapshots = true
db.settings = settings

With this change, timestamps stored in Cloud Firestore will be read back as
Firebase Timestamp objects instead of as system Date objects. So you will 
also need to update code expecting a Date to instead expect a Timestamp. 
For example:

// old: 
let date: Date = documentSnapshot.data["created_at"]
// new:
let timestamp: Timestamp = documentSnapshot.data["created_at"]
let date: Date = timestamp.approximateDateValue()

Please audit all existing usages of Date when you enable the new behavior. This new
behavior will become the default in a future release and so if you do not follow these
steps, YOUR APP MAY BREAK.
```

FYI- I didn't test my code snippets and my Swift is shaky, so they need validating. :-)

Alternatively, we could perhaps put this information in a doc page somewhere and link to it from this log message. This would give us the chance to improve / fix it if necessary, but also may mean it's less likely to be noticed and addressed. @samtstern, any thoughts?

Whatever we end up with here needs to be included in our official API Review, as on-by-default log messages are considered part of our API (and I think we want feedback on our breaking change migration strategy anyway).

Reply:
Strong warning in-place is always nice, but you can also add a goo.gl link to docs at the bottom/top that can point to a docs page. The reason for the short link is so that we can change where it goes if we move docs around.

Docs page has the advantage of knowing the current version, timelines, etc.

Reply:
Thanks Sam...  Is there a sane place to put this sort of temporary migration information in the docs?  Or would it basically be a dangling page that's only linked from this warning message (and probably release notes, etc.)?

Reply:
I think release notes is the right place.

Reply:
Michael, I added your suggestion as is for now, assuming it's non-final due to pending API review.